### PR TITLE
Zohaib/frontendassessment

### PR DESCRIPTION
**Pull request by zohaib68-adly**

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -25,8 +32,24 @@ export default function RootLayout({
...
+            src="/bacgrounds/topSectionBackground.png"
+            alt="Landing Page Background"
+            width={640}
+            height={269}
+            className="object-contain rounded-b-full opacity-50"
+            style={{
+              mixBlendMode: 'lighten',
+              background: 'radial-gradient(127.88% 53.54% at 50% 49.91%, rgba(24, 15, 79, 0) 0%, #240861 100%)'
+            }}
+            unoptimized
```
Comment (uxaircodes) [inline src/app/layout.tsx:49]: This disables next.js/sharp image optimization and is not recommended.

**Comment by uxaircodes** [comment]
**Reviewed the PR and noticed several key deviations from the defined scope and standards:**
- The task explicitly required React.js, but the app was built using Next.js. Tech stack decisions should align with what's outlined by the Product Owner and Leads.
- The assigned ticket should remain the single source of truth—implementation must follow the scope precisely.

**Technical concerns:**
- Atomic Design principles were not followed; business logic is mixed with components (e.g., inside LandingPage).
- Error boundaries are missing, which are critical for React error handling.
- There's no proper API/service layer; abstraction and error handling are lacking.
- Multiple type safety issues need resolution.

**Must Do**
- Prioritize fixing error handling and type safety, then address structural and scalability improvements.

There are significant gaps in the UI/UX implementation when compared to the Figma design. Several key design elements have not been properly implemented or are missing entirely.
A brief report outlining the major UI/UX issues has been provided in the attached PDF for reference.
[Zohaib-Assessment-Feedback.pdf](https://github.com/user-attachments/files/20991895/Zohaib-Assessment-Feedback.pdf)

**Comment by uxaircodes** [comment]
**PR Submission Guidelines**
Please also ensure that once you’ve completed the implementation of your assigned task, you perform a thorough QA pass before submitting the PR. This helps surface any overlooked issues, validates that the feature works as expected, and significantly improves the overall stability and readiness of the code for production.

**Additional best practices to follow before PR submission:**
- Run and pass all lint and build checks. Ensure no linting or compilation errors exist.
- Conduct end-to-end and edge case testing. Validate the flow across different scenarios, including unexpected or extreme cases.
- Cross-check with Figma designs. UI/UX should align with the design system and visual expectations.
- Test responsiveness. Make sure the UI works across various screen sizes if applicable.
- Self-review your code. Read through your own changes to catch any mistakes, unused code, or opportunities to improve readability.
- Verify integration points. Check API calls, state management, and any shared components for potential breakage.
- Clean up console logs and unused code. Avoid leaving debug statements or redundant code in your PR.

By following these practices, we ensure higher-quality deliveries, reduce review time, and promote smoother merges into the main branch.
